triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+      signatory owner
+
+  breedingRate : Int
+  breedingRate = 34
+
+  breedingPeriod : RelTime
+  breedingPeriod = seconds 1
+
+  breedingTrigger : Trigger (Bool, Int)
+  breedingTrigger = Trigger
+    { initialize = pure (False, 0)
+    , updateState = \msg -> case msg of
+        MHeartbeat -> do
+          (_, breedCount) <- get
+          put (True, breedCount + breedingRate)
+        _ -> do
+          (_, breedCount) <- get
+          put (False, breedCount)
+    , rule = \party -> do
+        (heartbeat, breedCount) <- get
+        if heartbeat then
+          forA_ [1..breedingRate] \offset -> do
+            void $ emitCommands [createCmd (Cat party (breedCount + offset))] []
+        else
+          pure ()
+    , registeredTemplates = RegisteredTemplates [ registeredTemplate @Cat ]
+    , heartbeat = Some breedingPeriod
+    }
+
+Wrapping Trigger Processes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Trigger processes have the Scala type ``Behavior[TriggerProcess.Message]`` and, once the Akka typed actor has been spawned they will have the type ``ActorRef[TriggerProcess.Message]``.
+
+Complex trigger process definitions may be defined by encapsulating instances of the spawned Akka typed actor ``ActorRef[TriggerProcess.Message]``. For example, given a Scala function ``transform: TriggerProcess.Message => TriggerProcess.Message`` we could write the following generic wrapper process:
+
+.. code-block:: scala
+  object TransformMessages {
+    def apply(
+      transform: TriggerProcess.Message => TriggerProcess.Message
+    )(
+      consumer: ActorRef[TriggerProcess.Message]
+    ): Behavior[TriggerProcess.Message] = {
+      Behaviors.receiveMessage { msg =>
+        consumer ! transform(msg)
+        Behaviors.same
+      }
+    }
+  }
+
+Alternatively, given a `Scalacheck <https://scalacheck.org>`_ generator ``Gen[TriggerProcess.Message]``, we could write the following wrapper process:
+
+.. code-block:: scala
+  object GeneratedMessages {
+    def apply(
+      msgGen: Gen[TriggerProcess.Message],
+      duration: FiniteDuration,
+    )(
+      consumer: ActorRef[TriggerProcess.Message]
+    ): Behavior[TriggerProcess.Message] = {
+      Behaviors.withTimers[Unit] { timer =>
+        timer.startTimerWithFixedDelay((), duration, duration)
+
+        Behaviors.receive { case (context, _) =>
+          msgGen.sample match {
+            case Some(msg) =>
+              consumer ! msg
+              Behaviors.same
+
+            case None =>
+              context.log.warn("Scalacheck generator stopped producing messages")
+              Behaviors.stopped
+          }
+        }
+      }
+    }
+  }
+
+In the following subsubsections, we present a number of pre-defined wrapper processes.
+
+.. note::
+  Wrapper processes allow complex and potentially non-standard (i.e. behaviours that are not easily definable in Daml code alone) trigger behaviours to be defined. This allows engineers to experiment with and research prototype trigger extensions.
```

Comment:
To introduce `behaviours` in the sentence before introducing it in the subclause `i.e.`,

```
Wrapper processes allow complex and potentially non-standard (i.e. behaviours that are not easily definable in Daml code alone) trigger behaviours to be defined.
```
can be changed to
```
Wrapper processes allows engineers to define complex and potentially non-standard trigger behaviours, i.e. behaviours that are not easily definable in Daml code alone.
```